src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticEETypeNode.cs

```diff
@@ -112,46 +83,22 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
             dataBuilder.Alignment = 16;
             dataBuilder.DefinedSymbols.Add(this);
 
-            bool hasPointers = NumSeries > 0;
-            if (hasPointers)
-            {
-                for (int i = ((_runLengths.Length / 2) * 2) - 1; i >= 0; i--)
-                {
-                    if (_targetPointerSize == 4)
-                    {
-                        dataBuilder.EmitInt(_runLengths[i]);
-                    }
-                    else
-                    {
-                        dataBuilder.EmitLong(_runLengths[i]);
-                    }
-                }
-                if (_targetPointerSize == 4)
-                {
-                    dataBuilder.EmitInt(NumSeries);
-                }
-                else
-                {
-                    dataBuilder.EmitLong(NumSeries);
-                }
-            }
+            int totalSize = _gcMap.Size * _target.PointerSize;
 
-            int totalSize = 0;
-            foreach (int run in _runLengths)
-            {
-                totalSize += run * _targetPointerSize;
-            }
+            GCDescEncoder.EncodeStandardGCDesc(ref dataBuilder, _gcMap, totalSize, 0);
+
+            Debug.Assert(dataBuilder.CountBytes == ((ISymbolNode)this).Offset);
 
             dataBuilder.EmitShort(0); // ComponentSize is always 0
 
-            if (hasPointers)
-                dataBuilder.EmitShort(0x20); // TypeFlags.HasPointers
-            else
-                dataBuilder.EmitShort(0x00);
+            dataBuilder.EmitShort((short)EETypeFlags.HasPointersFlag);
```

Comment:
Assuming we will use the current design for thread statics - where both GC and non-GC statics are allocated together - we may end up with EEType with no pointers here.